concrete/controllers/search/express/entries.php

```diff
@@ -27,28 +29,37 @@ public function search(Entity $entity)
-            $defaultSortColumn = $set->getDefaultSortColumn();
-            if ($this->request->query->has($this->entryList->getQuerySortDirectionParameter())) {
-                $direction = $this->request->query->get($this->entryList->getQuerySortDirectionParameter());
-            } else {
-                $direction = $defaultSortColumn->getColumnDefaultSortDirection();
-            }
-            if ($this->request->query->has($this->entryList->getQuerySortColumnParameter())) {
-                $value = $this->request->query->get($this->entryList->getQuerySortColumnParameter());
-                $column = $this->entity->getResultColumnSet();
-                $value = $column->getColumnByKey($value);
-                if (is_object($value)) {
-                    $this->entryList->sanitizedSortBy($value->getColumnKey(), $direction);
+        $app = Facade::getFacadeApplication();
+        $provider = $app->make(SearchProvider::class, ['entity' => $entity]);
+        $query = $provider->getSessionCurrentQuery();
+        if (is_object($query)) {
+            $this->result = $provider->getSearchResultFromQuery($query);
+            $this->result->setBaseURL(\URL::to('/ccm/system/search/express/entries/submit', $this->entity->getID()));
+        } else {
+
+            $set = $this->entity->getResultColumnSet();
+            if ($set) {
+                $defaultSortColumn = $set->getDefaultSortColumn();
+                if ($this->request->query->has($this->entryList->getQuerySortDirectionParameter())) {
+                    $direction = $this->request->query->get($this->entryList->getQuerySortDirectionParameter());
+                } else {
+                    $direction = $defaultSortColumn->getColumnDefaultSortDirection();
+                }
+                if ($this->request->query->has($this->entryList->getQuerySortColumnParameter())) {
+                    $value = $this->request->query->get($this->entryList->getQuerySortColumnParameter());
+                    $column = $this->entity->getResultColumnSet();
+                    $value = $column->getColumnByKey($value);
+                    if (is_object($value)) {
+                        $this->entryList->sanitizedSortBy($value->getColumnKey(), $direction);
+                    }
+                } else {
+                    $this->entryList->sanitizedSortBy($defaultSortColumn->getColumnKey(), $direction);
                 }
-            } else {
-                $this->entryList->sanitizedSortBy($defaultSortColumn->getColumnKey(), $direction);
-            }
 
-            $result = new Result($this->entity->getResultColumnSet(), $this->entryList,
-                \URL::to('/ccm/system/search/express/entries/submit', $this->entity->getID()));
-            $this->result = $result;
+                $result = new Result($this->entity->getResultColumnSet(), $this->entryList,
+                    \URL::to('/ccm/system/search/express/entries/submit', $this->entity->getID()));
```

Comment:
It's better to use the FQN to `\URL` which is `\Concrete\Core\Support\Facade\Url`, that is a likely candidate for class name collision